crates/sui-graphql-client/src/lib.rs

```diff
@@ -376,53 +379,57 @@ impl Client {
     // ===========================================================================
 
     /// Get the chain identifier.
-    pub async fn chain_id(&self) -> Result<String, Error> {
+    pub async fn chain_id(&self) -> Result<String, error::Error> {
         let operation = ChainIdentifierQuery::build(());
         let response = self.run_query(&operation).await?;
 
         if let Some(errors) = response.errors {
-            return Err(Error::msg(format!("{:?}", errors)));
+            return Err(QueryResponseError(errors).into());
         }
 
         response
             .data
             .map(|e| e.chain_identifier)
-            .ok_or_else(|| Error::msg("No data in response"))
+            .ok_or_else(|| EmptyResponse.into())
     }
 
     /// Get the reference gas price for the provided epoch or the last known one if no epoch is
     /// provided.
     ///
     /// This will return `Ok(None)` if the epoch requested is not available in the GraphQL service
     /// (e.g., due to pruning).
-    pub async fn reference_gas_price(&self, epoch: Option<u64>) -> Result<Option<u64>, Error> {
+    pub async fn reference_gas_price(
+        &self,
+        epoch: Option<u64>,
+    ) -> Result<Option<u64>, error::Error> {
         let operation = EpochSummaryQuery::build(EpochSummaryArgs { id: epoch });
         let response = self.run_query(&operation).await?;
 
-        if let Some(data) = response.data {
-            data.epoch
-                .and_then(|e| e.reference_gas_price.map(|x| x.try_into()))
-                .transpose()
-        } else if let Some(errors) = response.errors {
-            Err(Error::msg(format!("{:?}", errors)))
-        } else {
-            Err(Error::msg("No data in response"))
+        if let Some(errors) = response.errors {
+            return Err(QueryResponseError(errors).into());
         }
+
+        let d = response.data;
+
+        d.and_then(|e| e.epoch)
+            .and_then(|e| e.reference_gas_price)
+            .map(|x| x.try_into().map_err(|e: ParseError| e.into()))
+            .transpose()
     }
 
     /// Get the protocol configuration.
     pub async fn protocol_config(
         &self,
         version: Option<u64>,
-    ) -> Result<Option<ProtocolConfigs>, Error> {
+    ) -> Result<Option<ProtocolConfigs>, error::Error> {
```

Comment:
Another option would be to define a Result type and use crate::Result everywhere to avoid needing to specify the Error type in all these functions. You'll still need the error type in scope to do some of these things of course.